### ISSUE TITLE
ava.lang.RuntimeException: Couldn't find database maxwell

### ISSUE THREAD

**Issue by saimon7** [zendesk/maxwell#1172]
After updating Maxwell from 1.18.0 to 1.19.2 I have exception:

`
java.lang.RuntimeException: Couldn't find database maxwell
	at com.zendesk.maxwell.replication.TableCache.processEvent(TableCache.java:28) ~[classes/:?]
	at com.zendesk.maxwell.replication.BinlogConnectorReplicator.getTransactionRows(BinlogConnectorReplicator.java:391) ~[classes/:?]
	at com.zendesk.maxwell.replication.BinlogConnectorReplicator.getRow(BinlogConnectorReplicator.java:491) ~[classes/:?]
	at com.zendesk.maxwell.replication.BinlogConnectorReplicator.work(BinlogConnectorReplicator.java:149) ~[classes/:?]
	at com.zendesk.maxwell.util.RunLoopProcess.runLoop(RunLoopProcess.java:27) ~[classes/:?]
	at com.zendesk.maxwell.Maxwell.startInner(Maxwell.java:224) ~[classes/:?]
	at com.zendesk.maxwell.Maxwell.start(Maxwell.java:156) ~[classes/:?]
	at com.zendesk.maxwell.Maxwell.main(Maxwell.java:245) ~[classes/:?]
`

**config.properties:**

log_level=INFO

schema_database=maxwell
host=127.0.0.1
user=maxwell
password=pass
port=3306

replication_host=127.0.0.2
replication_user=maxwell_read
replication_password=pass
replication_port=3306

producer=stdout

blacklist_dbs=/^(?!my_replication_db).*$/

In my case I have `maxwell` and `my_replication_db` databases on the different MySQL host.

**Comment by osheroff** [comment]
oh... boy.  I think I see what's causing this.  In your case is `replication_host` a replica of `host`?  the `maxwell` data changes bypass the blacklist, but in your case we don't know about the maxwell database due to the split replica/schema roles.

**Comment by osheroff** [comment]
and was `replication_host` always a replica of `host`?

**Comment by osheroff** [comment]
i think I get why this is happening, working out a fix.

**Comment by czoIg** [comment]
I've got the same error (Couldn't find database maxwell at ...) but in following configuration:
Both host and replication_host are the same machine (127.0.0.1) and I've got two maxwell-dedicated users:
* read/write with only maxwell database visible
* read only with only my_replication_db visible

**Comment by mouhong** [comment]
I got the same error too.

My config: main_db to be replicated from and the schema_database internally used by maxwell are in two completely different servers (Adding a new maxwel db in production main db is usually not allowed). Any idea?

**Comment by osheroff** [comment]
@mouhong please describe your configuration more.  are the servers replicas of each other?  are you using blacklists?  please post a configuration sample and a backtrace of the error.

**Comment by mouhong** [comment]
@osheroff After some debugging in the source code just now, I think I got the reason: the replication_user must have read permission to **all** databases in the replication_host unless blacklisted.

Here's my setup:
- `replication_host` with 3 databases: db1, db2, db3, maxwell
- `replication_user` has read permission to db1, db2, db3
- `replication_user` has slave replication permission on `*.*`
- `user` has read/write permission to maxwell db

Because `replication_user` doesn't have access to maxwell db, the schema of maxwell db won't be captured. But `replication_user` will still receive replication events from maxwell db (they are still in the same server). When the events of maxwell db arrived, maxwell tried to find the schema in schema db, but it was not there, so it failed.

In short, I logically split the databases different groups with separate user permissions in one single server. That cause the problem. I think it would be better to highlight the requirements of replication_user's permission in the documentation. ;P

**Comment by mouhong** [comment]
I think it's the same cause to @czoIg 's question. His `replication_user` doesn't have access to the maxwell db in the same server while maxwell db is always whitelisted.
